hypothesis-python/src/hypothesis/stateful.py

```diff
@@ -606,12 +606,10 @@ def __init__(self, machine):
         )
 
     def do_draw(self, data):
-        try:
-            rule = data.draw(st.sampled_from(self.rules).filter(self.is_valid))
-        except HypothesisException:
-            # FailedHealthCheck or UnsatisfiedAssumption depending on user settings.
+        if not any(self.is_valid(rule) for rule in self.rules):
             msg = u"No progress can be made from state %r" % (self.machine,)
             quiet_raise(InvalidDefinition(msg))
+        rule = data.draw(st.sampled_from(self.rules).filter(self.is_valid))
```

Comment:
I don't know why this broke, but nor do I know what I was thinking to use exceptions here in the first place...